### PR TITLE
chore: collect windows cse logs by jumpbox

### PR DESCRIPTION
**Pull request by junjiezhang1997**
**What type of PR is this?**
/kind improvement

**What this PR does / why we need it**:
Collect cse logs by jumpbox to analyze the failure of node provisioning

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: e2e/jumpbox.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: jumpbox
```
Comment (AbelHu) [inline e2e/jumpbox.yaml:4]: why do we not use `az vmss run-command` to collect logs but with jumpbox? 
The node provisioning may fail before installing sshserver.

Comment (junjiezhang1997) [reply e2e/jumpbox.yaml:4]: I don't know how to set the command of powershell. When I use Get-Content, the length of return message is limited and I can only get the last several lines of the log file.

Comment (AbelHu) [reply e2e/jumpbox.yaml:4]: we can use `az vmss run-command` to upload the logs to a pre-created storage account

File: e2e/e2e-windows-scenario.sh [hunk 2]
```diff
@@ -8,15 +8,50 @@ choose() {
...
+    wget https://aka.ms/downloadazcopy-v10-linux
+    tar -xvf downloadazcopy-v10-linux
+    tokenWithoutQuote=${token//\"}
+    # use array to pass shellcheck
+    array=(azcopy_*)
+    ${array[0]}/azcopy copy "https://abe2ecselog.blob.core.windows.net/cselogs/${DEPLOYMENT_VMSS_NAME}-cse.log?${tokenWithoutQuote}" $SCENARIO_NAME-logs/CustomDataSetupScript.log
+    if [ "$retval" != "0" ]; then
+        echo "failed download cse logs"
+    fi 
+    ${array[0]}/azcopy rm "https://abe2ecselog.blob.core.windows.net/cselogs/${DEPLOYMENT_VMSS_NAME}-cse.log?${tokenWithoutQuote}"
```
Comment (AbelHu) [inline e2e/e2e-windows-scenario.sh:38]: Is it better to keep the remote cse log if we failed in downloading cse logs?

File: e2e/e2e-windows-scenario.sh [hunk 3]
```diff
@@ -8,15 +8,50 @@ choose() {
...
+        echo "failed upload cse logs"
+    fi
+    wget https://aka.ms/downloadazcopy-v10-linux
+    tar -xvf downloadazcopy-v10-linux
+    tokenWithoutQuote=${token//\"}
+    # use array to pass shellcheck
+    array=(azcopy_*)
+    ${array[0]}/azcopy copy "https://abe2ecselog.blob.core.windows.net/cselogs/${DEPLOYMENT_VMSS_NAME}-cse.log?${tokenWithoutQuote}" $SCENARIO_NAME-logs/CustomDataSetupScript.log
+    if [ "$retval" != "0" ]; then
+        echo "failed download cse logs"
```
Comment (AbelHu) [inline e2e/e2e-windows-scenario.sh:36]: failed in downloading?

File: e2e/e2e-windows-scenario.sh [hunk 4]
```diff
@@ -8,15 +8,50 @@ choose() {
...
+    fi 
+    ${array[0]}/azcopy rm "https://abe2ecselog.blob.core.windows.net/cselogs/${DEPLOYMENT_VMSS_NAME}-cse.log?${tokenWithoutQuote}"
+    if [ "$retval" != "0" ]; then
+        echo "failed delete cse logs in remote storage"
```
Comment (AbelHu) [inline e2e/e2e-windows-scenario.sh:40]: failed in deleting?

File: e2e/e2e-windows-scenario.sh [hunk 5]
```diff
@@ -8,15 +8,50 @@ choose() {
...
 }
 
+collect-logs() {
+    local retval
+    retval=0
+    mkdir -p $SCENARIO_NAME-logs
+    VMSS_INSTANCE_ID="$(az vmss list-instances --name $DEPLOYMENT_VMSS_NAME -g $MC_RESOURCE_GROUP_NAME | jq -r '.[0].instanceId')"
+    set +x
+    expiryTime=$(date --date="2 day" +%Y-%m-%d)
+    token=$(az storage container generate-sas --account-name abe2ecselog --account-key $MAPPED_ACCOUNT_KEY --permissions 'rwacdl' --expiry $expiryTime --name cselogs --https-only)
```
Comment (AbelHu) [inline e2e/e2e-windows-scenario.sh:18]: It is better not to hardcode storage account name in the code.

File: e2e/e2e-windows-scenario.sh [hunk 6]
```diff
@@ -8,15 +8,51 @@ choose() {
...
+        --parameters arg1=$DEPLOYMENT_VMSS_NAME arg2=$token
+    if [ "$retval" != "0" ]; then
+        echo "failed in uploading cse logs"
+    fi
+    wget https://aka.ms/downloadazcopy-v10-linux
+    tar -xvf downloadazcopy-v10-linux
+    tokenWithoutQuote=${token//\"}
+    # use array to pass shellcheck
+    array=(azcopy_*)
+    ${array[0]}/azcopy copy "https://abe2ecselog.blob.core.windows.net/cselogs/${DEPLOYMENT_VMSS_NAME}-cse.log?${tokenWithoutQuote}" $SCENARIO_NAME-logs/CustomDataSetupScript.log
```
Comment (AbelHu) [inline e2e/e2e-windows-scenario.sh:34]: do you plan to replace `abe2ecselog` with `$STORAGE_ACCOUNT_NAME` in all URLs?

Comment (junjiezhang1997) [reply e2e/e2e-windows-scenario.sh]: I think it's better to replace it in all URLs.  Besides, I modify the container name together.

**Review by AbelHu — APPROVED** [review]

**Pull request merged**